plugins/sparc_engage_warn.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+# Shows a warning message before engage the SPARC mirror
+'''
+Created on 3 Feb 2021
+
+@author: Éric Piel
+Copyright © 2021 Éric Piel, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms of the GNU
+General Public License version 2 as published by the Free Software Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+Public License for more details.
+
+You should have received a copy of the GNU General Public License along with Odemis. If not,
+see http://www.gnu.org/licenses/.
+'''
+
+from __future__ import division
+
+import logging
+from odemis.gui.cont.tabs import MIRROR_PARKED
+from odemis.gui.plugin import Plugin
+import wx
+
+ENGAGE_MSG = """Are you sure you want to engage the mirror?
+* Ensure the stage is at a safe working distance.
+* Ensure all detectors are retracted.
+"""
+
+PARK_MSG = """Are you sure you want to park the mirror?
+* Ensure all detectors are retracted.
+"""
+
+
+class SparcEngageWarnPlugin(Plugin):
+    name = "SPARC engage warn"
+    __version__ = "1.0"
+    __author__ = u"Éric Piel"
+    __license__ = "GPLv2"
+
+    def __init__(self, microscope, main_app):
+        super(SparcEngageWarnPlugin, self).__init__(microscope, main_app)
+        # Allow to run it on pretty much anything (excepted the viewer)
+        if not microscope or microscope.role != "sparc2":
```

Comment:
Better just check if the chamber tab is shown? Or use the same check as the chamber tab (the actual mirror stage is present).